src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -64,9 +80,113 @@ internal static string GetProtocolVersion(SafeSslHandle ssl)
+        internal static extern int SslShutdown(SafeSslHandle ssl);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void SslSetBio(SafeSslHandle ssl, SafeBioHandle rbio, SafeBioHandle wbio);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int SslDoHandshake(SafeSslHandle ssl);
+
+        [DllImport(Libraries.CryptoNative)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool IsSslStateOK(SafeSslHandle ssl);
+
         // NOTE: this is just an (unsafe) overload to the BioWrite method from Interop.Bio.cs.
         [DllImport(Libraries.CryptoNative)]
         internal static unsafe extern int BioWrite(SafeBioHandle b, byte* data, int len);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern SafeX509Handle SslGetPeerCertificate(SafeSslHandle ssl);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern SafeSharedX509StackHandle SslGetPeerCertChain(SafeSslHandle ssl);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int SslCtxUseCertificate(SafeSslContextHandle ctx, SafeX509Handle certPtr);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int SslCtxUsePrivateKey(SafeSslContextHandle ctx, SafeEvpPKeyHandle keyPtr);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int SslCtxCheckPrivateKey(SafeSslContextHandle ctx);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int BioCtrlPending(SafeBioHandle bio);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void SslCtxSetQuietShutdown(SafeSslContextHandle ctx);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "SslGetClientCAList")]
+        private static extern SafeSharedX509NameStackHandle SslGetClientCAListPrivate(SafeSslHandle ssl);
+
+        internal static SafeSharedX509NameStackHandle SslGetClientCAList(SafeSslHandle ssl)
+        {
+            Interop.Crypto.CheckValidOpenSslHandle(ssl);
+
+            SafeSharedX509NameStackHandle handle = SslGetClientCAListPrivate(ssl);
+
+            if (!handle.IsInvalid)
+            {
+                handle.SetParent(ssl);
+            }
+
+            return handle;
+        }
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void SslCtxSetVerify(SafeSslContextHandle ctx, [MarshalAs(UnmanagedType.FunctionPtr)] SslCtxSetVerifyCallback callback);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void SetEncryptionPolicy(SafeSslContextHandle ctx, EncryptionPolicy policy);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void SslCtxSetClientCAList(SafeSslContextHandle ctx, SafeX509NameStackHandle x509NameStackPtr);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void GetStreamSizes(out int header, out int trailer, out int maximumMessage);
```

Comment:
OpenSsl does not expose any macro/API for trailer unlike Secur32.dll on Windows. Since the code only returns constants, it never fails and hence return type is void